### ISSUE TITLE
Upload, Save Buttons Overlap Text in 1.7.0

### ISSUE THREAD

**Issue by miqrogroove** [humhub/humhub#4634]
When editing a post in 1.7.0, near the end of the last line the body text is covered by the upload and Save buttons.

**Comment by buddh4** [comment]
Can you add a screenshot?

**Comment by buddh4** [comment]
Ok, we probably should extend the padding:

![grafik](https://user-images.githubusercontent.com/9800854/99295364-9e61dc00-2845-11eb-80af-779630977a9a.png)

**Comment by buddh4** [comment]
Hmm its hard to find a perfect right padding for all languages. Maybe we should switch to someting like:

![grafik](https://user-images.githubusercontent.com/9800854/99297259-4d9fb280-2848-11eb-925b-b8a592e1ad5a.png)

**Comment by buddh4** [comment]
We should also make sure to facilitate the full editor width rather than using right padding.

**Comment by miqrogroove** [comment]
Buttons below text looks good at first.  I am concerned this might cause the keyboard to overlap the buttons on mobile devices.  If the cursor goes to the bottom of the visible area, do the buttons disappear off the bottom of the page?

**Comment by tinkat69** [comment]
Just to add that this is not new to 1.7 - was the same in earlier versions if the last line of text is complete i.e. runs to the right edge. Extending CSS ProseMirror padding-bottom by 20-30px seems to solve this. In cases with HH 1.6.4, "ProseMirror atwho-input form-control humhub-ui-richtext" padding-bottom has been increased from 6px to 30px (padding-right could then be decreased):
<img width="723" alt="re-text-upload-6pxpadding" src="https://user-images.githubusercontent.com/69628628/99541497-2ca6a100-29b1-11eb-9859-06ca3e3d23d8.png">
<img width="757" alt="re-text-upload-30pxpadding" src="https://user-images.githubusercontent.com/69628628/99541914-b6566e80-29b1-11eb-9322-72488699418e.png">

**Comment by buddh4** [comment]
I'd suggest moving the buttons beneath the whole input and only show them if the comment richtext is focused. This saves us many problems and complexity as:

- Fix the issue of this ticket
- Positioning style of upload/save buttons
- Repositioning when overflow (maxHeight) is activated, which currently requires a javascript handler
- Currently we can not use the resize browser feature (which would be great to have) since the buttons would overlap the resizer
- We would have 100% width and enhance usability
- We could add more buttons if wished

Regarding mobile, I think we should make sure to properly scroll into view in this case, which may already is the case.

**Comment by buddh4** [comment]
![grafik](https://user-images.githubusercontent.com/9800854/109048624-94b70280-76d7-11eb-9d04-d53a03f769bb.png)

![grafik](https://user-images.githubusercontent.com/9800854/109049132-2fafdc80-76d8-11eb-8949-b68069babda0.png)

**Comment by ArchBlood** [comment]
> ![grafik](https://user-images.githubusercontent.com/9800854/109057474-10b64800-76e2-11eb-893c-a07f4805d947.png)
> 
> ![grafik](https://user-images.githubusercontent.com/9800854/109056806-3858e080-76e1-11eb-9120-707fcdfe463b.png)
> 
> ![grafik](https://user-images.githubusercontent.com/9800854/109056997-7d7d1280-76e1-11eb-985b-a563593b9774.png)

I like these myself, it's more defined and eye catching :thinking:

**Comment by buddh4** [comment]
Those are the focus styles, only the editor menu is missing in here.

**Comment by luke-** [comment]
I would prefer this variant. 
https://user-images.githubusercontent.com/9800854/109049132-2fafdc80-76d8-11eb-8949-b68069babda0.png

Maybe we could try to display the submit button only when there is content in the editor.

**Comment by buddh4** [comment]
Yes, I'd only show the buttons either if the editor is focused or not empty.

**Comment by luke-** [comment]
We should use this variant: 

![image](https://user-images.githubusercontent.com/4736168/152356906-64edd4b9-f5b3-4e3a-bdae-a1e370afc7fc.png)

**Comment by luke-** [comment]
@yurabakhtin Thanks for the effort! We should really rethink that.

@orangiene : Can you please take over here again?

**Comment by orangiene** [comment]
@luke- @yurabakhtin 
I'll take a look and come back to you with another proposal.

**Comment by orangiene** [comment]
I have thought again about the problem with the languages. Is it really necessary to write "send" on the button? Can't we work with an icon as an alternative?
I think that our users are so smart, that's learned. On other platforms, the button is sometimes completely omitted.

That way we would have solved both problems: 
The user has a send button and the text field doesn't get any problems. In addition, we have more space for the actual use of the text field.

Here are three examples, with v02 using a modified button style.

I think v03 can be implemented well. What do you think?

![Comments-01](https://user-images.githubusercontent.com/62644500/153173750-fab8b74e-8901-4492-a1c4-77cf4bcc79fa.jpg)

![Comments-02](https://user-images.githubusercontent.com/62644500/153173827-15cfab3d-c0e0-4d5d-9a54-16d51c046bd9.jpg)

![Comments-03a](https://user-images.githubusercontent.com/62644500/153173917-b5ad23cb-486c-482d-aae7-e1d9edcea524.jpg)

![Comments-03b](https://user-images.githubusercontent.com/62644500/153174020-4945bdec-c602-4164-9c3a-d2401d4906d9.jpg)

**Comment by yurabakhtin** [comment]
@orangiene 

> I think v03 can be implemented well. What do you think? @luke- @yurabakhtin

I think v03 version will has same issue with long words as currently:

<img width="600" alt="long_word" src="https://user-images.githubusercontent.com/6995524/153185599-224d81a5-36a5-4259-9167-e0e45f1006f1.png">

It would be better if we move all buttons out the text field completely like on your v01 and v02.

**Comment by orangiene** [comment]
@yurabakhtin 
i thought maybe we could limit the text width in v03 so that there is always a 16px space between the text and the icon. (as drawn in pink)

But otherwise my favourite would be v02 ;-)

**Comment by aaukt** [comment]
I would prefer if both buttons are either inside or outside the editor and not one in the other one out.
Having them outside and below the editor would also leave the option to add more functions next to the upload, for example the emoji button or an "advanced formatting" button that only shows the menu bar when you actually need it.

**Comment by miqrogroove** [comment]
See related issue #3258 because it might be part of the same fix.

**Comment by yurabakhtin** [comment]
@luke- Can I start to implement v02?

![153173827-15cfab3d-c0e0-4d5d-9a54-16d51c046bd9](https://user-images.githubusercontent.com/6995524/154028893-c08ead09-b14e-4ef0-a59c-d5fb0532dfdd.jpeg)

**Comment by luke-** [comment]
@yurabakhtin Yes, please start with v2. Would be great if we could add also the FileHandlers (if available) to the File Upload Button.

**Comment by yurabakhtin** [comment]
@luke- 

> Would be great if we could add also the FileHandlers (if available) to the File Upload Button.

Ok, I will do this later - https://github.com/humhub/humhub/issues/5530#issuecomment-1040110956.

**Comment by yurabakhtin** [comment]
@orangiene @luke- PR https://github.com/humhub/humhub/pull/5562

<img width="956" alt="wide_screen" src="https://user-images.githubusercontent.com/6995524/154063144-77329791-489f-4d79-aa3f-f2f3f7007a70.png">

For narrow screens <= 414px I decided to put the buttons on next line:

<img width="447" alt="narrow_screen" src="https://user-images.githubusercontent.com/6995524/154063175-f344aa15-65ec-4a5e-a8f0-287e856804b5.png">

@luke- I detected a small issue:

<img width="960" alt="reply_issue" src="https://user-images.githubusercontent.com/6995524/154063235-eadbdb1c-cea1-47ca-a94b-235e06c4eaa3.png">

I will to fix this in the same PR.

> Would be great if we could add also the FileHandlers (if available) to the File Upload Button.

It will be done too in the same PR, or let me know if you prefer a new separate.

**Comment by luke-** [comment]
@yurabakhtin Looks nice for me. Feel free to bundle the changes into one PR.
@orangiene Can you please take a look into this tommorrow?

**Comment by yurabakhtin** [comment]
@luke- Commit https://github.com/humhub/humhub/pull/5562/commits/e72c44c1e29e1778ca8b4ae79cd470bf7fe30acc - for fix the issue with hidden reply form.

Commit https://github.com/humhub/humhub/pull/5562/commits/44663968a03f84a349d45d3bebd2472e4a4e4539 - for same style on edit post in stream:

<img width="671" alt="post_edit_form" src="https://user-images.githubusercontent.com/6995524/154067974-80e69f9c-aceb-4251-a1c3-e137657ed3f4.png">

**Comment by yurabakhtin** [comment]
@luke- Commit https://github.com/humhub/humhub/pull/5562/commits/b94ea06455d80da71518ae00ee1d8cf8efd456ed for issue https://github.com/humhub/humhub/issues/5530, but now I have done only for comment create/edit forms, for post edit form styles are broken and file is not attached correctly yet, I will fix this tomorrow.

<img width="886" alt="attach" src="https://user-images.githubusercontent.com/6995524/154094919-37e4d79d-c368-4c0c-bc23-05a3b6738c7e.png">

**Comment by yurabakhtin** [comment]
@luke- Commit https://github.com/humhub/humhub/pull/5562/commits/6566fc04679005e03812ff8463a43f79a87eb09c:

<img width="660" alt="attach_post_edit_form" src="https://user-images.githubusercontent.com/6995524/154222615-1577b10d-6973-4c4d-802e-4d7fe1be69e2.png">

I still need to fix tests, it seems the errors are related to new changes in the button "Send".

**Comment by yurabakhtin** [comment]
@luke- Commit https://github.com/humhub/humhub/pull/5562/commits/ac24d4110acfb084a9353f40c6bfc1a9fcc3e667 - fixed tests.

**Comment by luke-** [comment]
@yurabakhtin Awesome thanks!
